src/core/Api.js

```diff
@@ -0,0 +1,59 @@
+import fetch from 'node-fetch';
+
+export default class Api {
+  constructor(serverUrl) {
+    this._url = serverUrl;
+  }
+
+  request(route, method, url = this._url, body = {}) {
+    return new Promise(async (resolve, reject) => {
+      let result;
+      if (method === 'GET') {
+        result = await fetch(`${url}/${route}`, { method });
+      } else {
+        result = await fetch(`${url}/${route}`, { method, body });
+      }
+      const { status, ...data } = await result.json();
+      !status ? resolve(data) : reject(data);
+    });
+  }
+
+  listAllChains() {
+    return this.request('chains/', 'GET');
+  }
+
+  listAllSnapshots(chainType = 'ganache') {
+    return this.request(`snapshots/${chainType}`, 'GET');
+  }
+
+  getChain(id) {
+    return this.request(`chain/${id}`, 'GET');
+  }
+
+  deleteChain(id) {
+    return this.request(`chain/${id}`, 'DELETE');
+  }
+
+  downloadSnapshot(id) {
+    window.location.assign(`${this._url}/snapshot/${id}`);
```

Comment:
let's return a url and let the consuming frontend decide what to do instead of dealing with `window` from inside this library like this?